### ISSUE TITLE
Ray status not report correctly after node crashed

### ISSUE THREAD

**Issue by valiantljk** [ray-project/ray#15459]
### What is the problem?
I had a 3 nodes ray cluster on k8s, and ray is 1.2.0. Each node(POD) has 20cpu, 100 G memory

After submitting a task requesting 100 cpus, (I didn't enable autoscaler), I got error messages: tasks can not be placed, 
with 40 pending tasks. This is what I expected, but it also said the raylet might crashed bc of lost too many heartbeat. 
I checked the number alive pods, both worker pods are gone, only head pod is alive. 

However, on the ray head node, with 'ray status', it still shows 3 nodes, and 60 cpus. Seems the ray doesn't report correct ray cluster status. My questions are:
* why the code crash worker pods?
* why ray doesn't report cluster status correctly?

Detailed errors:
```
print_logs: [Errno 32] Broken pipe
The actor or task with ID ffffffffffffffff1f5f89da1a3796377df5cfb801000000 cannot be scheduled right now. It requires {CPU: 1.000000} for placement, but this node only has remaining {0.000000/20.000000 CPU, 175.683594 GiB/175.683594 GiB memory, 1.000000/1.000000 node:192.168.13.7, 51.953125 GiB/51.953125 GiB object_store_memory}
. In total there are 0 pending tasks and 40 pending actors on this node. This is likely due to all cluster resources being claimed by actors. To resolve the issue, consider creating fewer actors or increase the resources available to this Ray cluster. You can ignore this message if this Ray cluster is expected to auto-scale.





The node with node id f69930a5f31cb116298e16c76073d93fb2bef506203356e5557eb8f8 has been marked dead because the detector has missed too many heartbeats from it. This can happen when a raylet crashes unexpectedly or has lagging heartbeats.
The node with node id b774c3d3727676fd4060891394830038751b93e29c7d45c7ed12bf6a has been marked dead because the detector has missed too many heartbeats from it. This can happen when a raylet crashes unexpectedly or has lagging heartbeats.
```
*Ray version and other system information (Python version, TensorFlow version, OS):*
Ray: 1.2.0
Python: 3.7
OS: Linux
### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):
The code is basically from ray torchtrainer examples, and run as:

```
python example.py num_workers num_batch mode(0/1/2, 0: remote, 1: connect to existing ray, 2: init new ray cluster)
e.g.,
python example.py 100 1 1 # I used remote mode in my test, which python example.py 100 1 0
```
 
```
import ray
from ray.util.sgd import TorchTrainer
from ray.util.sgd.torch import TrainingOperator
from ray.util.sgd.torch.examples.train_example import LinearDataset

import torch
import logging,sys
from torch.utils.data import DataLoader

class CustomTrainingOperator(TrainingOperator):
    def setup(self, config):
        # Load data.
        train_loader = DataLoader(LinearDataset(2, 5), config["batch_size"])
        val_loader = DataLoader(LinearDataset(2, 5), config["batch_size"])

        # Create model.
        model = torch.nn.Linear(1, 1)

        # Create optimizer.
        optimizer = torch.optim.SGD(model.parameters(), lr=1e-2)

        # Create loss.
        loss = torch.nn.MSELoss()

        # Register model, optimizer, and loss.
        self.model, self.optimizer, self.criterion = self.register(
            models=model,
            optimizers=optimizer,
            criterion=loss)

        # Register data loaders.
        self.register_data(train_loader=train_loader, validation_loader=val_loader)

#class 

if __name__ == "__main__":
    logging.basicConfig(level=logging.INFO)
    if len(sys.argv) < 4:
        logging.warning("python pytorch.py Num_worker Epochs 0/1/2")
        logging.warning("0: remote, 1: existing, 2: new")
        exit() 

    connect_mode = int(sys.argv[3])

    if connect_mode == 0:
        try:    
            ray.util.connect("xxxxxxx:xxx")
        except Exception as e:
            logging.error(f"remote connection failed:{e}")
    if connect_mode == 1:
        try:
            ray.init(address="auto") # connect to existing cluster 
        except Exception as e:
            logging.error(f"in cluster connection failed:{e}")
    if connect_mode == 2:
        try:
            ray.init()
        except Exception as e:
            logging.error(f"new cluster connection failed:{e}")
    num_workers = int(sys.argv[1])
    epochs = int(sys.argv[2])
    logging.info(f"number of workers is:{num_workers}")
    logging.info(f"number of epochs is:{epochs}")

    trainer1 = TorchTrainer(
        training_operator_cls=CustomTrainingOperator,
        num_workers=num_workers,
        use_gpu=False,
        config={"batch_size": 64})

    for _ in range(epochs):
        stats = trainer1.train()
        print(stats)
    trainer1.shutdown()
    print("success!")
```

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by richardliaw** [comment]
@wuisawesome can you take a look at this?

**Comment by wuisawesome** [comment]
@DmitriGekhtman can you take a look at this too?

**Comment by chaomengyuan** [comment]
Similar observation found even though my setting up is a little bit different.

I have a cluster with 3 nodes, each node has 48 CPU and 104 GiB memory.

I am using Ray version 2.0.0.dev0 and testing Ray autoscaler with local cluster using _python/ray/autoscaler/local/example-minimal-manual.yaml_ to launch the cluster. I put one node's ip as the _head_ip_ and the other two's ips as _worker_ips_. Then, I added the following configuration _min_workers: 0_, _max_workers: 2_ and _idle_timeout_minutes: 1_ into the yaml file.

After I launch the Ray cluster with _ray up_, I run _ray status_ and I see the head node is running. Then, I execute a small piece of code below which needs 120 CPUs.

```
import ray
import time

ray.init(address='auto')

@ray.remote
def increment():
    value = 0
    for i in range(30):
        value += 1
        time.sleep(1)
    return value


print(ray.get([increment.remote() for _ in range(120)]))
```

Then, when I run _ray status_, I can see that the healthy nodes increasing from 1 to 3 and then back to 1 after idle for 1 min. However, the resources usage information keeps to be as follows even if the healthy node becomes 1 again.

> Resources
> ---------------------------------------------------------------
> 
> Usage:
>  0.0/144.0 CPU
>  0.00/314.617 GiB memory
>  0.00/138.827 GiB object_store_memory

My question is: why does the resources usage information not get updated as the healthy node information?

**Comment by DmitriGekhtman** [comment]
@chaomengyuan
The reason is that we don't properly support min_worker != max_workers for local (on-prem) clusters (and we should document that better).

What's happening internally is that the autoscaler is marking the two workers as not part of the cluster but they are not being disconnected from Ray.

I will open a separate issue for that -- if dynamic autoscaling on your on-prem cluster is something you require, please comment there so we can take appropriate action.

**Comment by DmitriGekhtman** [comment]
> @DmitriGekhtman can you take a look at this too?

K8s support was in pretty bad shape in Ray 1.2.0
Would be good if the problem was confirmed in master or latest release (Ray 1.4.1)

**Comment by DmitriGekhtman** [comment]
@chaomengyuan and/or @AmeerHajAli please check if https://github.com/ray-project/ray/pull/17062 correctly resolves the scale-down issue
(I'm aware both of you have straightforward access to on-prem clusters, which I do not.)

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
